pkg/rke2/rc.go

```diff
@@ -0,0 +1,85 @@
+package rke2
+
+import (
+	"context"
+	"sync"
+
+	"github.com/k3s-io/k3s/pkg/cli/cmds"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+const runtimeClassesChart = "rke2-runtimeclasses"
+
+var runtimes = map[string]bool{
+	"nvidia":              true,
+	"nvidia-experimental": true,
+	"crun":                true,
+}
+
+func setRuntimes() cmds.StartupHook {
+	return func(ctx context.Context, wg *sync.WaitGroup, args cmds.StartupHookArgs) error {
+		go func() {
+			defer wg.Done()
+			<-args.APIServerReady
+			logrus.Info("Setting runtimes")
+
+			config, err := clientcmd.BuildConfigFromFlags("", args.KubeConfigSupervisor)
+			if err != nil {
+				logrus.Fatalf("runtimes: new k8s restConfig: %v", err)
+			}
+
+			client, err := kubernetes.NewForConfig(config)
+			if err != nil {
+				logrus.Fatalf("runtimes: new k8s client: %v", err)
+			}
```

Comment:
We should probably fix this in the other files in pkg/rke2 as well - but you can use https://github.com/k3s-io/k3s/blob/master/pkg/util/client.go#L30
```suggestion
			client, err := util.GetClientSet(args.KubeConfigSupervisor)
			if err != nil {
				logrus.Fatalf("runtimes: new k8s client: %v", err)
			}
```